### PR TITLE
Kmeans

### PR DESCRIPTION
**Pull request by PrimozGodec**
Implemented complete k-means add-on

### CODE REVIEW

File: orangecontrib/educational/widgets/owkmeans.py [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+    icon = "icons/mywidget.svg"
+    want_main_area = False
+
+    # inputs and outputs
+    inputs = [("Data", Orange.data.Table, "set_data")]
+    outputs = [("Annotated Data", Table, widget.Default),
+               ("Centroids", Table)]
+
+    # settings
+    numberOfClusters = settings.Setting(1)
```
Comment (astaric) [inline orangecontrib/educational/widgets/owkmeans.py]: underscore_casing?

File: orangecontrib/educational/widgets/owkmeans.py [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+    lines_to_centroids = settings.Setting(0)
+    graph_name = 'scatter'
+    outputName = "cluster"
+    button_labels = {"step1": "Reassign membership",
+                     "step2": "Recompute centroids",
+                     "step_back": "Step back",
+                     "autoplay_run": "Run",
+                     "autoplay_stop": "Stop",
+                     "random_centroids": "Randomize"}
+    colors = ['#2f7ed8', '#0d233a', '#8bbc21', '#910000', '#1aadce',
```
Comment (astaric) [inline orangecontrib/educational/widgets/owkmeans.py]: How did you define this colors? What to do if more are needed?

File: orangecontrib/educational/widgets/owkmeans.py [hunk 3]
```diff
@@ -0,0 +1,532 @@
...
+                                   tooltip_shared=False,
+                                   debug=True)  # TODO: set false when end of development
+        # Just render an empty chart so it shows a nice 'No data to display'
+        self.scatter.chart()
+        self.mainArea.layout().addWidget(self.scatter)
+
+    def concat_x_y(self):
+        """
+        Function takes two selected columns from data table and merge them in new Orange.data.Table
+        :return: table with selected columns
```
Comment (astaric) [inline orangecontrib/educational/widgets/owkmeans.py]: We tend to prefer numpy-style napoleon docstrings. You can tell PyCharm to use different style in Python Integrated Tools section of the Preferences.

File: orangecontrib/educational/widgets/owkmeans.py [hunk 4]
```diff
@@ -0,0 +1,532 @@
...
+
+    def restart(self):
+        """
+        Function triggered on data change or restart button pressed
+        """
+        self.k_means = Kmeans(self.concat_x_y())
+        self.modify_kmeans()
+
+    def modify_kmeans(self):
+
```
Comment (astaric) [inline orangecontrib/educational/widgets/owkmeans.py]: This line could be removed.

File: orangecontrib/educational/widgets/owkmeans.py [hunk 5]
```diff
@@ -0,0 +1,532 @@
...
+    def complete_replot(self):
+        """
+        This function performs complete replot of the graph without animation
+        """
+        attr_x, attr_y = self.data.domain[self.attr_x], self.data.domain[self.attr_y]
+
+        # plot centroids
+        options = dict(series=[])
+        options['series'].append(dict(data=[{'x': p[0],
```
Comment (astaric) [inline orangecontrib/educational/widgets/owkmeans.py]: If you make a new line before data=, you can reduce the empty space on the left.

File: orangecontrib/educational/widgets/owkmeans.py (hunk 2 above)
Comment (PrimozGodec) [inline orangecontrib/educational/widgets/owkmeans.py]: More is never required because number of clusters is limited to 10.

**Pull request merged**